react/components/Spinner/index.js

```diff
@@ -5,62 +5,91 @@ import { baseClassname } from '../icon/utils'
 const radius = 40
 const circ = 2 * radius * Math.PI
 
-const Spinner = ({ color, size, block }) => (
-  <svg
-    className={
-      `${baseClassname('spinner')} ${!color ? 'c-action-primary' : ''} ${block ? 'db' : ''}`
-    }
-    xmlns="http://www.w3.org/2000/svg"
-    viewBox="0 0 100 100"
-    preserveAspectRatio="xMidYMid"
-    height={size}
-    width={size}
-  >
-    <circle
-      cx="50"
-      cy="50"
-      fill="none"
-      r={radius}
-      stroke={color || 'currentColor'}
-      strokeWidth="10"
-      strokeDasharray={`0 0 2 ${circ}`}
-      strokeLinecap="round"
-      strokeDashoffset="1"
-      transform="rotate(96 50 50)"
-    >
-      <animate
-        attributeName="stroke-dasharray"
-        dur="1.5s"
-        repeatCount="indefinite"
-        values={
-          `0 0 2 ${circ};
-          0 0 ${circ * 0.75} ${circ};
-          0 ${circ - 2} ${circ * 0.75} ${circ}`
+class Spinner extends React.Component {
+  constructor(props) {
+    super(props)
+    this.spinner = React.createRef()
+  }
+
+  componentDidMount(props) {
+    const element = this.spinner.current
+
+    element.addEventListener('webkitAnimationIteration', () => {
+      console.log('Completed')
+    })
+    element.addEventListener('animationIteration', () => {
+      console.log('Completed')
+    })
```

Comment:
@lbebber I added some event here to see if it was working, but like you said the `<animation>` is a different API, so it didn't.

This PR is just transforming the Spinner into a class in case we'll need to edit the lifecycle methods. Feel free to use it or discard.